packages/procaptcha-pow/src/services/Manager.ts

```diff
@@ -126,140 +128,129 @@ export const Manager = (
 	};
 
 	const start = async () => {
-		if (state.loading) {
-			return;
-		}
-		if (state.isHuman) {
-			return;
-		}
-
-		resetState();
-
-		// set the loading flag to true (allow UI to show some sort of loading / pending indicator while we get the captcha process going)
-		updateState({
-			loading: true,
-		});
-
-		const config = getConfig();
-
-		// check if account exists in extension
-		const ext = config.web2 ? new ExtensionWeb2() : new ExtensionWeb3();
-
-		// use the passed in account (could be web3) or create a new account
-		const userAccount =
-			config.userAccountAddress ||
-			(await ext.getAccount(config)).account.address;
-
-		// set the account created or injected by the extension
-		updateState({
-			account: { account: { address: userAccount } },
-		});
-
-		// snapshot the config into the state
-		updateState({ dappAccount: config.account.address });
-
-		// allow UI to catch up with the loading state
-		await sleep(100);
-
-		// check if account has been provided in config (doesn't matter in web2 mode)
-		if (!config.web2 && !config.userAccountAddress) {
-			throw new ProsopoEnvError("GENERAL.ACCOUNT_NOT_FOUND", {
-				context: { error: "Account address has not been set for web3 mode" },
-			});
-		}
-
-		// get a random provider
-		const getRandomProviderResponse = getRandomActiveProvider();
-
-		const events = getDefaultEvents(onStateUpdate, state, callbacks);
+		await providerRetry(
```

Comment:
I don't think this needs to be a fn which wraps everything. Imo it should do the provider retry at provider selection, i.e on the getRandomActiveProvider call. It can stay as is for now, but I think this should be simplified at some point bc it's rather confusing atm

Reply:
Resolved on slack